src/wrappers/gcc_wrapper.hpp

```diff
@@ -39,9 +46,11 @@ class gcc_wrapper_t : public program_wrapper_t {
   string_list_t get_input_files() override;
   std::string preprocess_source() override;
   string_list_t get_implicit_input_files() override;
+  virtual bool uses_defines_in_preprocess() const;
```

Comment:
This is an extension of the "wrapper API" for GCC class wrappers. Please document it with doxygen like the other virtual functions (see https://github.com/mbitsnbites/buildcache/blob/master/src/wrappers/program_wrapper.hpp#L55 and forward), and put a blank line between this method the other overridden method, like this:

```
  ...
  string_list_t get_implicit_input_files() override;

  // @brief Bla bla bla...
  // ...
  virtual bool uses_defines_in_preprocess() const;
```